File: week2/greet.c
#include    <stdio.h>
#include    <cs50.h>

int main(int argc, string argv[]) //now the main function has two arguments: argc and argv
{
    if (argc == 2)  //this is how you check how much the user has typed in a command line argument, in this case 2, the name of the program (argc) and the argument (argv).
        printf("hello, %s\n", argv[1]); //this is how you print the command line argument
    }
    else
    {
        printf("hello, world\n");
    }
}

File: week6/greet.py
from sys import argv        # Import argv from sys module

if len(argv) == 2:          # If there are 2 command-line arguments (the first is the program name)
    print(f"Hello, {argv[1]}")  # Print the second argument
else:                       # Otherwise
    print("Hello, world")    # Print "Hello, world"

 #this function will print the second command line argument, and if there is none, it will print "Hello, world"